src/Neo/SmartContract/Native/LedgerContract.cs

```diff
@@ -107,6 +113,9 @@ public UInt256 GetBlockHash(DataCache snapshot, uint index)
         [ContractMethod(CpuFee = 1 << 15, RequiredCallFlags = CallFlags.ReadStates)]
         public UInt256 CurrentHash(DataCache snapshot)
         {
+            if (snapshot is null)
+                throw new ArgumentNullException(nameof(snapshot));
+
```

Comment:
More error details are always welcomed, but is it ever possible that `snapshot` is null in this context? I mean, not in unit-tests, but in real life. It's the contract method, hence `snapshot` here is not a user-provided argument; it's the system-provided argument:
https://github.com/neo-project/neo/blob/6e000529d7efb0a7fd589e162a03a88cef799415/src/Neo/SmartContract/Native/NativeContract.cs#L389
So to me in this context `snapshot` can never be null.